### PR TITLE
[TTAHUB-605] Add filter count to filters button

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

Add filter count to filters button.

## How to test

Go to any page that uses the new filters. If a filter is selected by default this should show in the filter button text as "Filters (x)". Adding and removing filters should also update this number.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-605

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
~- [ ] API Documentation updated~
~- [ ] Boundary diagram updated~
~- [ ] Logical Data Model updated~
~- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

**Comment by jasalisbury** [comment]
Screenshot from 2022-02-23 12-27-30
We'll want to not update the number until "apply" is clicked

**Review by kryswisnaskas — APPROVED** [review]
Looks good! 👍

File: frontend/src/components/filter/FilterMenu.js [hunk 1]
```diff
@@ -183,6 +183,7 @@ export default function FilterMenu({
       canBlur={canBlur}
       AlternateActionButton={ClearAllButton}
       onOpen={onOpen}
+      filterCount={filters.length}
```
Comment (jasalisbury) [inline frontend/src/components/filter/FilterMenu.js:186]: Can we update the `buttonText` prop adding the number there? Doing it that way means we don't have to pass a prop down to the dropdown menu and can keep the dropdown menu more generic (not having to worry about filters, even though I see a `test-id` that references a filter that was in there before this PR)

Comment (AdamAdHocTeam) [reply frontend/src/components/filter/FilterMenu.js]: Updated to use button text.

File: frontend/src/components/filter/FilterMenu.js [hunk 2]
```diff
@@ -171,7 +171,7 @@ export default function FilterMenu({
 
   return (
     <DropdownMenu
-      buttonText="Filters"
+      buttonText={`Filters${filters && filters.length > 0 ? ` (${filters.length})` : ''}`}
       buttonAriaLabel="open filters for this page"
```
Comment (jasalisbury) [inline frontend/src/components/filter/FilterMenu.js:175]: Do we need to add the number of filters to the aria label as well?

**Review by jasalisbury — APPROVED** [review]
:rocket:

File: frontend/src/components/filter/FilterMenu.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/filter/FilterMenu.js]: This has been updated.

**Pull request merged**